app/models/user.rb

```diff
@@ -19,4 +21,34 @@ class User < ApplicationRecord
   def downcase_email
     email.downcase
   end
+
+  class << self
+    def digest string
+      cost =
+        if ActiveModel::SecurePassword.min_cost
+          BCrypt::Engine::MIN_COST
+        else
+          BCrypt::Engine.cost
+        end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attributes remember_digest: User.digest(remember_token)
```

Comment:
dạ update_attribute thì sẽ bỏ qua validation còn update_attributes thì không nên dùng update_attributes em nghi an toàn hơn ạ